app/Nova/Actions/ExportResumes.php

```diff
@@ -0,0 +1,172 @@
+            return Action::danger('Sorry! You are not authorized to perform this action.');
+        }
+
+        $users = User::active()
+            ->whereNotNull('resume_date')
+            ->where('resume_date', '>', $fields->resume_date_cutoff)
+            ->where('primary_affiliation', 'student')
+            ->whereDoesntHave('duesPackages', static function (Builder $q): void {
+                $q->where('restricted_to_students', false);
+            })
+            ->leftJoin('major_user', static function (JoinClause $join): void {
+                $join->on('users.id', '=', 'major_user.user_id')
+                     ->whereNull('major_user.deleted_at');
+            })
+            ->leftJoin(
+                'majors',
+                'major_user.major_id',
+                '=',
+                'majors.id'
+            )
+            ->where('majors.display_name', $fields->major)
+            ->orderBy('last_name')
+            ->orderBy('first_name')
+            ->pluck('uid');
+
+        if (0 === $users->count()) {
+            return Action::danger('No resumes matched the criteria!');
+        }
+
+        $filenames = $users->map(static function (string $uid): string {
+            return escapeshellarg(Storage::disk('local')->path('resumes/'.$uid.'.pdf'));
+        });
+
+        $datecode = now()->format('Y-m-d-Hi');
+        $filename = 'robojackets-resumes-'.Str::slug($fields->major).'-'.$datecode.'.pdf';
+        $path = Storage::disk('local')->path('nova-exports/'.$filename);
+
+        $coverfilename = 'robojackets-resumes-'.Str::slug($fields->major).'-'.$datecode.'-cover.pdf';
+        $coverpath = Storage::disk('local')->path('nova-exports/'.$coverfilename);
+
+        PDF::loadView(
+            'resumecover',
+            [
+                'major' => $fields->major,
+                'cutoff_date' => $fields->resume_date_cutoff,
+                'generation_date' => $datecode,
+            ]
+        )->save($coverpath);
+
+        // Ghostscript: -q -dNOPAUSE -dBATCH for disabling interactivity, -sDEVICE= for setting output type, -dSAFER
+        // because the input is untrusted
+        $cmd = 'gs -q -dNOPAUSE -dBATCH -sDEVICE=pdfwrite -dSAFER -sOutputFile='.escapeshellarg($path).' ';
+        $cmd .= $coverpath.' ';
+        $cmd .= $filenames->join(' ');
+
+        Log::debug('Running shell command: '.$cmd);
+        $gsOutput = [];
+        $gsExit = -1;
+        exec($cmd, $gsOutput, $gsExit);
+
+        if (0 !== $gsExit) {
+            Log::error('gs did not exit cleanly (status code '.$gsExit.'), output: '.implode("\n", $gsOutput));
+
+            return Action::danger('Error exporting resumes');
+        }
+
+        // This is not perfect! The original metadata is recoverable (exiftool can't remove it permanently).
+        $cmdExif = 'exiftool -Title="RoboJackets Resumes - '.
+            $fields->major.'" -Creator="MyRoboJackets" -Author="RoboJackets" ';
```

Comment:
Does Nova validate `$fields->major` is from the dropdown? This may leave us vulnerable to injection otherwise.

Reply:
I added a validation rule that I believe should prevent this.